Mathlib/Topology/ContinuousMap/Interval.lean

```diff
@@ -0,0 +1,148 @@
+/-
+Copyright (c) 2024 Vincent Beffara. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vincent Beffara
+-/
+import Mathlib.Topology.CompactOpen
+import Mathlib.Topology.Order.ProjIcc
+
+/-!
+# Continuous bundled maps on intervals
+
+In this file we prove a few results about `ContinuousMap` when the domain is an interval.
+-/
+
+open Set ContinuousMap Filter Topology
+
+namespace ContinuousMap
+
+variable {α : Type*} [LinearOrder α] [TopologicalSpace α] [OrderTopology α]
+variable {a b c : α} [Fact (a ≤ b)] [Fact (b ≤ c)]
+variable {E : Type*} [TopologicalSpace E]
+
+/-- The embedding into an interval from a sub-interval lying on the left, as a `ContinuousMap`. -/
+def subintervalLeft : C(Icc a b, Icc a c) :=
+  .inclusion <| Icc_subset_Icc le_rfl Fact.out
+
+/-- The embedding into an interval from a sub-interval lying on the right, as a `ContinuousMap`. -/
+def subintervalRight : C(Icc b c, Icc a c) :=
+  .inclusion <| Icc_subset_Icc Fact.out le_rfl
+
+/-- The map `projIcc` from `α` onto an interval in `α`, as a `ContinuousMap`. -/
+def projIccCM : C(α, Icc a b) :=
+  ⟨projIcc a b Fact.out, continuous_projIcc⟩
+
+/-- The extension operation from continuous maps on an interval to continuous maps on the whole
+  type, as a `ContinuousMap`. -/
+def IccExtendCM : C(C(Icc a b, E), C(α, E)) where
+  toFun f := f.comp projIccCM
+  continuous_toFun := continuous_precomp projIccCM
+
+@[simp]
+theorem IccExtendCM_of_mem {f : C(Icc a b, E)} {x : α} (hx : x ∈ Icc a b) :
+    IccExtendCM f x = f ⟨x, hx⟩ := by
+  simp [IccExtendCM, projIccCM, projIcc, hx.1, hx.2]
+
+/-- The concatenation of two continuous maps defined on adjacent intervals. If the values of the
+functions on the common bound do not agree, this is defined as an arbitrarily chosen constant
+map. See `concatCM` for the corresponding map on the subtype of compatible function pairs. -/
+noncomputable def concat (f : C(Icc a b, E)) (g : C(Icc b c, E)) :
+    C(Icc a c, E) := by
+  by_cases hb : f ⊤ = g ⊥
+  · let h (t : α) : E := if t ≤ b then IccExtendCM f t else IccExtendCM g t
+    suffices Continuous h from ⟨fun t => h t, by fun_prop⟩
+    apply Continuous.if_le (by fun_prop) (by fun_prop) continuous_id continuous_const
+    rintro x rfl
+    simpa [IccExtendCM, projIccCM]
+  · exact .const _ (f ⊥) -- junk value
+
+variable {f : C(Icc a b, E)} {g : C(Icc b c, E)}
+
+theorem concat_comp_left (hb : f ⊤ = g ⊥) :
+    f = (concat f g).comp subintervalLeft := by
+  ext x
+  simp [concat, IccExtendCM, hb, subintervalLeft, projIccCM, inclusion, x.2.2]
+
+theorem concat_comp_right (hb : f ⊤ = g ⊥) :
+    g = (concat f g).comp subintervalRight := by
```

Comment:
```suggestion
theorem concat_comp_subintervalRight (hb : f ⊤ = g ⊥) :
    (concat f g).comp subintervalRight = g := by
```

Reply:
OK that makes sense. I am only using that lemma for rewrites in the direction I had, but the simp direction is the one you are suggesting.